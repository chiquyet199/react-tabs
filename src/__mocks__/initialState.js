export const initialState = {
  locale: {
    currentLocale: "en-US",
    translations: {
      common: {
        postcode_lookup: {
          postcode: "Postcode",
          find_address: "Find address",
          address_not_listed: "Address not listed",
        },
      },
      home__welcome: "Welcome to RNW starter app",
      home__loggedin: "You are now logged in",
      login__email__label: "Email address",
      login__password__label: "Password",
      login__email__placeholder: "example@example.com",
      login__password__placeholder: "******",
      login__continue__with__facebook: "Continue with Facebook",
      login: "Sign in",
      create__account: "Create account",
      login__forgot__password: "Forgot password",
      logout: "Logout",
      back: "Back",
      submit: "Submit",
      rent_passport: "Rent Passport",
      deposit_free: "Deposit Free",
      home_management: "Home Management",
      authorize: "Authorize",
      terms_header: "Canopy terms & conditions",
      terms:
        "1. INTRODUCTION\n\nBy signing up for an account with Canopy, you are agreeing to be bound by the following terms and conditions (“Terms of Service”). These Terms of Service are to be read in conjunction with:\n\nour Privacy Policy, which sets out the terms on which we process any personal data we collect from you, or that you or others provide to us; and\n\nour Cookies Policy, which sets out information about the cookies we use.\n\nThese Terms of Service are important. Please read them carefully as they contain details of our statutory and regulatory responsibilities to you, and your contractual obligations to us. They also set out the basis on which we provide, and you are permitted to use, the various services made available on the website www.findyourcanopy.com and the mobile application known as ‘Canopy’ (together, the “Platform”), including the following:",
      verify__email__title__text: "Verify your email address",
      verify__email__text:
        "We need to know you are you before creating a RentPassport. Please click the link in the email we sent you.",
      open__email__client: "Open my email client",
      resend__verification__email: "Resend verification email",
      not__received__an__email: "Not received an email?",
      forgot_password: "Forgot your password?",
      reset_password: "Change your password",
      tenant_reference_submitted: "Reference submitted",
      tenant_reference_thanks_for_time:
        "Thanks for taking the time to submit a tenant reference for",
      tenant_reference_did_you_know:
        "Did you know that Canopy offers the following property management tools to tenants, landlords and agents:",
      tenant_reference_rent_passport_text:
        "Full tenant referencing and property management including deposits and payments",
      tenant_reference_deposit_free_text:
        "Qualified tenants can pay a premium to cover their rental security deposit",
      tenant_reference_home_management_text:
        "With a valid UK Open Banking account, tenants are able to have rent payments contribute to their Experian credit report",
      find_out_more: "Find out more",
      tenant_reference_title: "Tenant reference request",
      upload_tenant_reference_document: "Upload tenant reference document",
      upload_tenant_reference_document_description:
        "If you have a manual form you usually use, please upload it here.",
      upload: "Upload",
      submit_reference: "Submit reference",
      tenant_reference_request: "Tenant reference request",
      complete_reference_form_description:
        " tenant of yours, has authorised us to request a reference from you.\n" +
        "As you know a landlord reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
      upload_reference_document: "Upload reference document",
      complete_reference_form: "Complete reference form",
      complete_reference_form_a: ", a ",
      truelayer: {
        truelayer_auth_title: "Truelayer Authorize",
      },
      employer_reference_form_description:
        " has advised us that you are their employer and has authorised us to request a reference from you.\n" +
        "As you know an employment reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
    },
  },
  auth: {
    user: null,
    isLoggedIn: false,
  },
  dimensions: {
    fontScale: 1,
    height: 901,
    scale: 1,
    width: 1920,
    size: "L",
  },
  inputField: {
    showPassword: true,
  },
  form: {},
  partialForm: {
    values: {},
  },
  progress: 0,
  notification: {
    type: null,
    message: "",
  },
  user: {
    profile: {
      firstName: "FirstName",
      lastName: "LastName",
    },
  },
  router: {
    location: {
      pathname: "/tenant-reference-request/complete",
      search: "",
      hash: "",
    },
    action: "POP",
  },
  lookups: {
    addresses: [],
  },
  topbar: {
    haveTopbar: true,
    topbarBtn: {
      haveTopbarButton: false,
      topbarButtonText: "",
      btnType: "transparent_green",
      topbarButtonIcon: undefined,
      hideBackButton: false,
    },
    topbarCenter: {
      topbarCenterText: "Center",
    },
    topbarLeft: {
      topbarLeftText: "Back",
    },
  },
  images: {
    accepted: [],
  },
  rentPassport: {
    aboutYou: {
      email: "jonathan@gmail.com",
      firstName: "Jonathan",
      lastName: "Beaumont",
      middleName: null,
      gender: "MALE",
      dayOfBirth: 20,
      monthOfBirth: 11,
      yearOfBirth: 1970,
      maritalStatus: "MARRIED",
      phoneNumber: "01123456789",
      nationality: "UK",
      isSmoker: true,
      hasPets: true,
      hasChildren: true,
      status: "GREEN",
      messages: [],
    },
    financial: {
      salary: 100000,
      pension: 0,
      interest: 0,
      benefits: 0,
      rentals: 0,
      other: 0,
      grossAnnualIncome: 100000,
      sourcesOfIncome: ["Employment"],
      status: "YELLOW",
      messages: [],
      bankrupt: false,
      creditScore: 930,
      depositFree: true,
      maximumEligible: 1800.266,
      activeCCJs: 0,
      activeCCJYear: 0,
      activeCCJMonth: 0,
      activeCCJValue: 0,
      previousCCJs: 0,
      previousCCJsYear: 0,
      previousCCJsMonth: 0,
      activeInsolvencies: 0,
      activeInsolvenciesYear: 0,
      activeInsolvenciesMonth: 0,
      previousInsolvencies: false,
      previousInsolvenciesYear: 0,
      previousInsolvenciesMonth: 0,
    },
    rentPassportLegal: {
      pep: false,
      mortality: "passed",
      legalWarnings: false,
      sanctions: false,
      status: "GREEN",
      messages: [],
    },
    whatYouDo: {
      workInfo: [
        {
          workStatus: "EMPLOYED",
          companyName: "companyName",
          jobTitle: "jobTitle",
          income: 100000,
          employmentTime: "FULL_TIME",
          employmentType: "CONTRACT",
          workedSinceMonth: 12,
          workedSinceYear: 2017,
          studentType: "POST_GRADUATE_STUDENT",
          reference: {
            contactFirstName: "Silvester",
            contactLastName: "Stallone",
            contactEmailAddress: "rambo666@gmail.com",
            referredAt: "12-12-2016",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
    whereYouLived: {
      residenceInfo: [
        {
          line1: "Addresss line 1",
          town: "Town",
          state: "State",
          postCode: "UL11223",
          countryCode: "UK",
          residenceType: "RENTING_FROM_AGENT",
          livedSinceMonth: 1,
          livedSinceYear: 2016,
          livedToMonth: null,
          livedToYear: null,
          reference: {
            contactFirstName: "Jonathan",
            contactLastName: "Beaumont",
            contactEmailAddress: "jonathan@gmail.com",
            referredAt: "01-01-2018",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
  },
};
