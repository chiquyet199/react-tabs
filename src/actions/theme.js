import {UPDATE_GLOBAL_THEME} from "./types";

export const updateTheme = globaltheme => ({
  type: UPDATE_GLOBAL_THEME,
  globaltheme,
});
