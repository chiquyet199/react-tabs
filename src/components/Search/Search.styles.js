import styledNative from "styled-components/native";
import {themed} from "../../utils/common";

export const SearchBox = styledNative.View`
  display: flex;
  padding: 11px 0;
  flex-direction: row;
`;

export const SearchInput = themed(styledNative.TextInput`
  font-size: 18px;
  border: 0;
  opacity: 0.6;
  line-height: 1.33;
  margin-left: 8px;
  &::placeholder {
    color: black;
    font-style: italic;
  }
  &:focus {
    outline-width: 0;
  }
`);
