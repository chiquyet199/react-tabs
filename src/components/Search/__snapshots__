/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Search/__native_snapshots__