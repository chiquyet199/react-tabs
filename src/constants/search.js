export const WAIT_INTERVAL = 1000;
export const ENTER_KEY = 13;
