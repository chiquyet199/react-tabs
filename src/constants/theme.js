import {isIos, isX} from "../utils/common";
export const padding = {
  top: (isIos && ((isX && 40) || 20)) || 0,
};
export const colors = {
  // shades of grey
  white: "#ffffff",
  black: "#000000",
  grey: "#6e7c91",
  canopySteel: "#6b7c93", // hq color
  canopySoftSteel: "#8999ab",
  steel: "#b9c7d6",
  lightSteel: "#e3eaf2",
  canopySoftSteelOverlay: "#f3f7fc",
  lightBackground: "rgba(255, 255, 255, 0.2)",
  shadowColor: "rgba(107, 124, 147, 0.4)",
  semiTransparent: "rgba(0, 0, 0, 0.12)",
  darkOverlay: "rgba(0, 0, 0, 0.8)",
  darkTransparent: "rgba(0, 0, 0, 0.7)",
  none: "rgba(0, 0, 0, 0)",
  primary: "#5ab88e",
  hqBlue: "#5ab1b8", // hq color
  lightBlue: "#42c8ce", // hq color
  lightTeal: "#37b3ba", // hq color for 'body link'
  veryLightSteel: "#f6f9fc", // hq color
  blueGray: "#455363", // hq colour
  blueGreen: "#3c7087",

  facebook: "#3b5998",
  dark: "#010202",
  canopyGreen: "#5ab88e",
  darkGreen: "#236a6d",
  darkestGreen: "#011e24", // not in zep, used to make gradient
  green: "#398d7f",
  lightGreen: "#58dfa3",
  successGreen: "#5dc471",
  darkConcrete: "#455363",
  whiteTransparent: "rgba(255, 255, 255, 0.2)",

  red: "#e26e59",
  alertRed: "#f16651",
  warning: "#fffcf3",
  orange: "#f1bf51",
  rentPassportErrorBg: "#fff5f5",
  rentPassportWarningBg: "#fffcf3",
};

const _switch = {
  accentColor: colors.successGreen,
  baseColor: colors.white,
};

const buttons = {
  primary: {
    background: colors.canopyGreen,
    border: colors.canopyGreen,
    text: colors.white,
    fontWeight: "bold",
  },
  secondary: {
    background: colors.white,
    border: colors.canopyGreen,
    text: colors.canopyGreen,
    fontWeight: "bold",
  },
  tertiary: {
    background: colors.white,
    border: colors.grey,
    text: colors.grey,
    fontWeight: "bold",
  },
  primary_outlined: {
    background: colors.none,
    border: colors.canopyGreen,
    text: colors.canopyGreen,
    fontWeight: "bold",
  },
  facebook: {
    background: colors.facebook,
    border: colors.facebook,
    text: colors.white,
    fontWeight: "bold",
  },
  danger: {
    background: colors.red,
    border: colors.red,
    text: colors.white,
    fontWeight: "bold",
  },
  danger_outlined: {
    background: colors.white,
    border: colors.red,
    text: colors.red,
    fontWeight: "bold",
  },
  confirmation_danger: {
    background: colors.white,
    border: colors.none,
    text: colors.red,
    fontWeight: "bold",
    fontSize: "18px",
  },
  confirmation_primary: {
    background: colors.white,
    border: colors.none,
    text: colors.hqBlue,
    fontWeight: "bold",
    fontSize: "18px",
  },
  confirmation_cancel: {
    background: colors.white,
    border: colors.none,
    text: colors.canopySteel,
    fontWeight: "bold",
    fontSize: "18px",
  },
  transparent: {
    background: colors.none,
    border: colors.none,
    text: colors.white,
    fontWeight: "bold",
  },
  transparent_green: {
    background: colors.none,
    border: colors.none,
    fontWeight: "bold",
    text: colors.canopyGreen,
    fontSize: "14px",
  },
  transparent_gray: {
    background: colors.none,
    border: colors.none,
    text: colors.canopySteel,
    fontWeight: "bold",
  },
  transparent_white: {
    background: colors.none,
    border: colors.none,
    text: colors.white,
    fontWeight: "bold",
  },
  tabButton: {
    background: colors.white,
    border: colors.lightSteel,
    text: colors.canopySteel,
    fontWeight: "bold",
    fontSize: "14px",
  },
  delete_button: {
    background: colors.white,
    border: colors.steel,
    color: colors.steel,
    fontWeight: "bold",
  },
};

export default {
  hq: {
    light: {
      background: [colors.white, colors.veryLightSteel],
      form: {
        accentColor: colors.hqBlue,
        baseColor: colors.grey,
      },
      topbar: {
        background: colors.white,
        baseColor: colors.canopySteel,
        headerButtonColour: "transparent_blue",
        progressBar: colors.lightBlue,
      },
      tabs: {
        background: colors.white,
        borderBottom: colors.hqBlue,
        activeText: colors.hqBlue,
        inactiveText: colors.canopySteel,
        indicator: colors.hqBlue,
        label: colors.hqBlue,
      },
      switch: _switch,
      accentColor: colors.hqBlue,
      baseColor: colors.canopySteel,
      textColor: colors.canopySteel,
      altTextColor: colors.white,
      activeTabColor: colors.hqBlue,
      inactiveTabColor: colors.lightSteel,
      dividerColor: colors.lightSteel,
      lineBreakColor: colors.lightSteel,
      errorColor: colors.red,
      boxBorder: colors.lightSteel,
    },
    dark: {
      background: [colors.white, colors.white],
      form: {
        accentColor: colors.hqBlue,
        baseColor: colors.grey,
      },
      topbar: {
        background: colors.none,
        baseColor: colors.white,
        headerButtonColour: "transparent_blue",
        progressBar: colors.lightBlue,
      },
      switch: _switch,
      accentColor: colors.hqBlue,
      baseColor: colors.white,
      textColor: colors.white,
      activeTabColor: colors.hqBlue,
      inactiveTabColor: colors.lightSteel,
      dividerColor: colors.lightSteel,
      lineBreakColor: colors.lightSteel,
    },
    buttons: {
      ...buttons,
      primary: {
        background: colors.hqBlue,
        border: colors.hqBlue,
        text: colors.white,
        fontWeight: "bold",
      },
      secondary: {
        background: colors.white,
        border: colors.hqBlue,
        text: colors.hqBlue,
        fontWeight: "bold",
      },
      outlined: {
        background: colors.none,
        border: colors.hqBlue,
        text: colors.hqBlue,
        fontWeight: "bold",
      },
      site_page_util_bar_primary: {
        background: colors.hqBlue,
        border: colors.hqBlue,
        fontWeight: "bold",
        text: colors.white,
      },
      site_page_util_bar_outlined: {
        background: colors.white,
        border: colors.hqBlue,
        fontWeight: "bold",
        text: colors.hqBlue,
      },
      transparent_blue: {
        background: colors.none,
        border: colors.none,
        fontWeight: "bold",
        text: colors.hqBlue,
      },
      transparent_green: {
        background: colors.none,
        border: colors.none,
        fontWeight: "bold",
        text: colors.hqBlue,
        fontSize: "14px",
      },
    },
    colors,
    padding,
  },
  selva: {
    dark: {
      background: [colors.darkGreen, colors.darkestGreen],
      form: {
        accentColor: colors.canopyGreen,
        baseColor: colors.white,
        bounceButton: "transparent",
      },
      topbar: {
        background: colors.none,
        baseColor: colors.white,
        headerButtonColour: "transparent_white",
        progressBar: colors.lightGreen,
      },
      tabs: {
        background: colors.white,
        borderBottom: colors.canopyGreen,
        activeText: colors.canopyGreen,
        inactiveText: colors.canopySteel,
        indicator: colors.canopyGreen,
        label: colors.canopyGreen,
      },
      switch: _switch,
      accentColor: colors.canopyGreen,
      baseColor: colors.veryLightSteel,
      textColor: colors.veryLightSteel,
      dividerColor: colors.lightSteel,
    },
    light: {
      background: [colors.white, colors.white],
      form: {
        accentColor: colors.canopyGreen,
        baseColor: colors.canopySteel,
        bounceButton: "transparent_green",
      },
      topbar: {
        background: colors.white,
        baseColor: colors.canopySteel,
        headerButtonColour: "transparent_green",
        progressBar: colors.lightGreen,
        fontSize: 18,
      },
      tabs: {
        background: colors.white,
        borderBottom: colors.canopyGreen,
        activeText: colors.canopyGreen,
        inactiveText: colors.canopySteel,
        indicator: colors.canopyGreen,
        label: colors.canopyGreen,
      },
      switch: _switch,
      accentColor: colors.canopyGreen,
      baseColor: colors.canopySteel,
      textColor: colors.canopySteel,
      errorTextColor: colors.alertRed,
      dividerColor: colors.lightSteel,
    },
    colors,
    buttons,
    padding,
  },
  hqApp: {
    light: {
      background: [colors.white, colors.white],
      form: {
        accentColor: colors.hqBlue,
        baseColor: colors.grey,
      },
      topbar: {
        background: colors.darkConcrete,
        baseColor: colors.white,
        fontSize: 18,
        headerButtonColour: "transparent_white",
        progressBar: colors.lightBlue,
      },
      tabs: {
        background: colors.white,
        borderBottom: colors.hqBlue,
        activeText: colors.hqBlue,
        inactiveText: colors.canopySteel,
        indicator: colors.hqBlue,
        label: colors.hqBlue,
      },
      switch: _switch,
      accentColor: colors.hqBlue,
      baseColor: colors.canopySteel,
      textColor: colors.canopySteel,
      boxBorder: colors.lightSteel,
      altBoxBorder: colors.veryLightSteel,
      lightBackground: colors.veryLightSteel,
      disabledTextColor: colors.steel,
      altTextColor: colors.white,
      activeTab: colors.canopyGreen,
      inactiveTab: colors.lightSteel,
      dividerColor: colors.lightSteel,
      dangerText: colors.red,
      successText: colors.successGreen,
      rentStatusTab: {
        green: colors.successGreen,
        yellow: colors.orange,
        red: colors.red,
      },
      selectedBoxBorder: colors.lightBlue,
      slideoutMenu: {
        background: colors.blueGreen,
      },
      activeTenancy: {
        propertySummaryBackground: colors.lightBackground,
      },
      lineBreakColor: colors.canopySteel,
      errorColor: colors.red,
    },
    dark: {
      background: [colors.darkConcrete, colors.darkConcrete],
      form: {
        accentColor: colors.hqBlue,
        baseColor: colors.grey,
      },
      topbar: {
        background: colors.darkConcrete,
        baseColor: colors.white,
        fontSize: 18,
        headerButtonColour: "transparent_white",
      },
      switch: _switch,
      accentColor: colors.hqBlue,
      baseColor: colors.white,
      textColor: colors.white,
      boxBorder: colors.lightSteel,
      lightBackground: colors.veryLightSteel,
      disabledTextColor: colors.steel,
      altTextColor: colors.white,
      activeTab: colors.canopyGreen,
      inactiveTab: colors.lightSteel,
    },
    buttons: {
      ...buttons,
      primary: {
        background: colors.hqBlue,
        border: colors.hqBlue,
        text: colors.white,
        fontWeight: "bold",
      },
      primary_outlined: {
        background: colors.none,
        border: colors.hqBlue,
        text: colors.hqBlue,
        fontWeight: "bold",
      },
      secondary: {
        background: colors.white,
        border: colors.hqBlue,
        text: colors.hqBlue,
        fontWeight: "bold",
      },
      outlined: {
        background: colors.none,
        border: colors.hqBlue,
        text: colors.hqBlue,
        fontWeight: "bold",
      },
    },
    colors,
    padding,
  },
};
