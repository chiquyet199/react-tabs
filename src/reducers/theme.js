import {UPDATE_GLOBAL_THEME} from "../actions/types";

const initialState = "light";

export default (state = initialState, {type, globaltheme}) => {
  switch (type) {
    case UPDATE_GLOBAL_THEME:
      return globaltheme;
    default:
      return state;
  }
};
