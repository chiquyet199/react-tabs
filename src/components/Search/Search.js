import React, {Component} from "react";
import {SearchInput, SearchBox} from "./Search.styles";
import {IconButton} from "../Button";
import {string, func} from "prop-types";
import {WAIT_INTERVAL, ENTER_KEY} from "../../constants/search";
import {withTheme} from "styled-components";

export const ThemedIconButton = withTheme(IconButton);

export class Search extends Component {
  state = {
    inputValue: "",
  };
  /**
   * time-slicing implementation
   * todo: redo this when react will support time-slicing out of the box
   *
   */
  timer = null;
  handleInputChange = event => {
    clearTimeout(this.timer);
    this.setState({inputValue: event.target.value});
    this.timer = setTimeout(this.triggerChange, WAIT_INTERVAL);
  };
  /**
   * handle user press enter
   * send fetch data request immediately
   *
   */
  handleKeyDown = event => {
    if (event.keyCode === ENTER_KEY) {
      clearTimeout(this.timer);
      this.triggerChange();
    }
  };
  triggerChange = () => {
    const {setFilterQuery} = this.props;
    setFilterQuery(this.state.inputValue);
  };
  render() {
    const {placeholder} = this.props;
    const {inputValue} = this.state;

    return (
      <SearchBox>
        <ThemedIconButton
          name="search"
          type="transparent_gray"
          iconsize={36}
          onClick={this.triggerChange}
        />
        <SearchInput
          type="search"
          placeholder={placeholder}
          onChange={this.handleInputChange}
          value={inputValue}
          onKeyDown={this.handleKeyDown}
        />
      </SearchBox>
    );
  }
}
Search.propTypes = {
  placeholder: string,
  setFilterQuery: func.isRequired,
};

export default Search;
