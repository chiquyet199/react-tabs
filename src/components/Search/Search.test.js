import React from "react";
import {Search, ThemedIconButton} from "./Search";
import {shallow} from "enzyme";
import configureStore from "redux-mock-store";
import {initialState} from "../../__mocks__/initialState";
import {Platform} from "react-native";
import {SearchInput} from "./Search.styles";
import theme from "../../constants/theme";
import {extend} from "underscore";
import {ENTER_KEY, WAIT_INTERVAL} from "../../constants/search";

const exampleValue = "port";
jest.useFakeTimers();
const mockStore = configureStore();

describe("Search component", () => {
  it("Search snapshot test on - " + Platform.OS, () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });
  describe("Search Input functionality", () => {
    it("passes the right arguments to `setFilterQuery`", async () => {
      const setFilterQuery = jest.fn();
      const component = getComponent({setFilterQuery});
      const input = component.find(SearchInput);
      const searchButton = component.find(ThemedIconButton);
      input.simulate("change", {target: {value: exampleValue}});
      searchButton.simulate("click");
      expect(setFilterQuery).toHaveBeenCalledWith(exampleValue);
    });
    describe("when onChange is called on the Search Input", () => {
      it("updates the search input value", () => {
        const component = getComponent();
        component
          .find(SearchInput)
          .simulate("change", {target: {value: exampleValue}});
        expect(component.find(SearchInput).props().value).toBe(exampleValue);
      });
      it("fires `setFilterQuery` after pending call to the trigger", () => {
        const component = getComponent();
        component
          .find(SearchInput)
          .simulate("change", {target: {value: exampleValue}});
        expect(setTimeout).toHaveBeenCalled();
        expect(setTimeout).toHaveBeenCalledWith(
          component.instance().triggerChange,
          WAIT_INTERVAL,
        );
      });
      it("calls `setFilterQuery` only once with the arguments provided by the second call", () => {
        jest.useFakeTimers();
        const setFilterQuery = jest.fn();
        const exampleValue2 = "starboard";
        const component = getComponent({setFilterQuery});
        const onChange = component.find(SearchInput).prop("onChange");
        onChange({target: {value: exampleValue2}});
        onChange({target: {value: exampleValue}});
        jest.runAllTimers();
        expect(setFilterQuery).toHaveBeenCalledTimes(1);
        expect(setFilterQuery).toHaveBeenCalledWith(exampleValue);
      });
      it("triggers search when user presses `Enter` on his keyboard", () => {
        const setFilterQuery = jest.fn();
        const component = getComponent({setFilterQuery});
        component
          .find(SearchInput)
          .simulate("change", {target: {value: exampleValue}});
        component.find(SearchInput).simulate("keyDown", {keyCode: ENTER_KEY});
        expect(setFilterQuery).toHaveBeenCalledWith(exampleValue);
      });
    });
  });
  const getComponent = props => {
    const parsedProps = extend(
      {
        placeholder: "A placeholder",
        theme,
        setFilterQuery: () => {},
      },
      props,
    );

    return shallow(<Search {...parsedProps} />, {
      context: {store: mockStore(initialState)},
    });
  };
});
