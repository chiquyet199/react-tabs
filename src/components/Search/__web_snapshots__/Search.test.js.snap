// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Search snapshot test on - web 1`] = `
<Styled(View)>
  <WithTheme(IconButton)
    iconsize={36}
    name="search"
    onClick={[Function]}
    type="transparent_gray"
  />
  <Connect(WithTheme(Styled(TextInput)))
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="A placeholder"
    type="search"
    value=""
  />
</Styled(View)>
`;
